codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/SerializeMiddleware.java

```diff
@@ -0,0 +1,223 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen.protocol.aws;
+
+import static software.amazon.smithy.go.codegen.GoStackStepMiddlewareGenerator.createSerializeStepMiddleware;
+import static software.amazon.smithy.go.codegen.GoWriter.emptyGoTemplate;
+import static software.amazon.smithy.go.codegen.GoWriter.goTemplate;
+import static software.amazon.smithy.go.codegen.server.protocol.JsonSerializerGenerator.getSerializerName;
+
+import java.util.Map;
+import software.amazon.smithy.go.codegen.EventStreamGenerator;
+import software.amazon.smithy.go.codegen.GoStdlibTypes;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SmithyGoTypes;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.ProtocolUtils;
+import software.amazon.smithy.go.codegen.trait.BackfilledInputOutputTrait;
+import software.amazon.smithy.model.knowledge.EventStreamIndex;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.utils.MapUtils;
+
+
+
+
+
+@SuppressWarnings("checkstyle:RegexpSingleline")
+public class SerializeMiddleware implements GoWriter.Writable {
+    protected final ProtocolGenerator generator;
+    protected final ProtocolGenerator.GenerationContext ctx;
+    protected final OperationShape operation;
+    protected final GoWriter writer;
+
+    protected final StructureShape input;
+    protected final StructureShape output;
+    private final EventStreamIndex eventStreamIndex;
+
+//    private final String SMITHY_PROTOCOL_NAME = "awsjson10";
+    private final String contentType = "application/awsjson10";
+    private String serialName;
+
+    public SerializeMiddleware(
+            ProtocolGenerator generator, ProtocolGenerator.GenerationContext ctx, OperationShape operation,
+            GoWriter writer
+    ) {
+        this.generator = generator;
+        this.ctx = ctx;
+        this.operation = operation;
+        this.writer = writer;
+
+        this.input = ctx.getModel().expectShape(operation.getInputShape(), StructureShape.class);
+        this.output = ctx.getModel().expectShape(operation.getOutputShape(), StructureShape.class);
+        this.eventStreamIndex = EventStreamIndex.of(ctx.getModel());
+    }
+
+    @Override
+    public void accept(GoWriter writer) {
+        var name = ProtocolGenerator.getSerializeMiddlewareName(operation.getId(), ctx.getService(),
+                generator.getProtocolName());
+        var middleware = createSerializeStepMiddleware(name, ProtocolUtils.OPERATION_SERIALIZER_MIDDLEWARE_ID);
+//        writer.write(middleware.asWritable(generateOperationStubs(), emptyGoTemplate()));
+    }
+
+    public GoWriter.Writable generateOperationStubs() {
```

Comment:
This is doing more than what the name implies (generating "stubs") - you're actually generating the whole middleware, so name it accordingly.

Reply:
e.g. "generate", "generateMiddleware", or writer.write() it directly as a Writable.